PowerFGT/Public/cmdb/system/interface.ps1

```diff
@@ -98,3 +98,276 @@ function Get-FGTSystemInterface {
     End {
     }
 }
+
+function Set-FGTSystemInterface {
+
+    <#
+        .SYNOPSIS
+        Set a vlan interface
+
+        .DESCRIPTION
+        Set a vlan interface
+
+        .EXAMPLE
+        Set-FGTSystemInterface -name PowerFGT -alias ALIAS_PowerFGT -role lan -mode static -address_mask 192.0.2.1/255.255.255.0  -admin_access ping,https -device_identification $false -connected $false
+
+        This set the interface vlan named PowerFGT with an alias, the LAN role, in static mode with 192.0.2.1 as IP, with ping and https administrative access, and with device identification disable and not connected
+    #>
+
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'medium')]
+    Param(
+        [Parameter (Mandatory = $true, Position = 1)]
+        [string]$name,
+        [Parameter (Mandatory = $false)]
+        [string]$alias,
+        [Parameter (Mandatory = $false)]
+        [ValidateSet('lan', 'wan', 'dmz', 'undefined')]
+        [string]$role,
+        [Parameter (Mandatory = $false)]
+        [ValidateSet('https', 'ping', 'fgfm', 'capwap', 'ssh', 'snmp', 'ftm', 'radius-acct', 'ftm')]
+        [string[]]$admin_access,
+        [Parameter (Mandatory = $false)]
+        [ValidateSet('static', 'dhcp')]
+        [string]$mode,
+        [Parameter (Mandatory = $false)]
+        [string]$address_mask,
+        [Parameter (Mandatory = $false)]
+        [string]$connected = $false,
+        [Parameter (Mandatory = $false)]
+        [string]$device_identification = $false,
+        [Parameter(Mandatory = $false)]
+        [String[]]$vdom,
+        [Parameter(Mandatory = $false)]
+        [psobject]$connection = $DefaultFGTConnection
+    )
+
+    Begin {
+    }
+
+    Process {
+
+        $invokeParams = @{ }
+        if ( $PsBoundParameters.ContainsKey('vdom') ) {
+            $invokeParams.add( 'vdom', $vdom )
+        }
+
+        $uri = "api/v2/cmdb/system/interface/$name"
+        $_interface = new-Object -TypeName PSObject
+
+        $_interface | add-member -name "name" -membertype NoteProperty -Value $name
+        $_interface | add-member -name "role" -membertype NoteProperty -Value $role
+        $_interface | add-member -name "mode" -membertype NoteProperty -Value $mode
+
+        if ( $PsBoundParameters.ContainsKey('alias') ) {
+            $_interface | add-member -name "alias" -membertype NoteProperty -Value $alias
+        }
+
+        if ( $PsBoundParameters.ContainsKey('admin_access') ) {
+            $allowaccess = [string]$admin_access
+            $_interface | add-member -name "allowaccess" -membertype NoteProperty -Value $allowaccess
+        }
+
+        if ( $PsBoundParameters.ContainsKey('address_mask') ) {
+            $_interface | add-member -name "ip" -membertype NoteProperty -Value $address_mask
+        }
+
+        switch ($connected) {
```

Comment:
use UP/Down and not switch

Reply:
what do you mean with this? do you want to use a string "up" / "down" as values instead of a boolean?

Reply:
Yes @jelmerj ! (and set connected by default...)